### PR TITLE
Introduce Anthropic module (+ bump to LangChain4j 0.30)

### LINKED ISSUE #413: Introduce Anthropic module
This will require work in the upstream LangChain4J project so we can introduce our own `AnthropicClient.Builder` (in the same way we do for Mistral & OpenAI).

### PR DESCRIPTION
**Pull request by edeandrea**
Introduce Anthropic module

> [!WARNING]
> This will not build until langchain4j/langchain4j#873 is merged, released, and then that release incorporated into this PR.
>
> I've built that PR locally and run this locally against that and all the tests pass. I'll move this PR out of draft status once langchain4j/langchain4j#873 is merged, released, and then that release incorporated into this PR.

Closes #413

### CODE REVIEW

**Comment by geoand** [comment]
This might need a rebase because the mistral docs issue was fixed in `main` in the meantime

**Comment by edeandrea** [comment]
Will do. I'll rebase it once the next upstream release happens.

**Comment by edeandrea** [comment]
According to [the latest comment](https://github.com/langchain4j/langchain4j/pull/873#issuecomment-2045082864) it looks like the next LangChain4j release which contains the fix I'm blocked on will be released on April 15.

**Comment by geoand** [comment]
Seems like [upstream](https://github.com/langchain4j/langchain4j/pull/897) now has tools support as well

**Comment by edeandrea** [comment]
I saw. https://github.com/langchain4j/langchain4j/pull/873 still hasn't been released, so we'll see. If they release that without the upstream tools support I may just finish this off so we can get it out the door, then go back to the tools thing.

I don't have much time this week to work on this, and then I'm out on PTO next week, so I'd rather get something out rather than holding on this for another several weeks.

Unless of course you'd rather wait.

**Comment by edeandrea** [comment]
Although looking at the PR it looks like there may be some breaking method changes. If that is released long with my other one at the same time I may be forced to re-work it into this.

We'll see once the next langchain4j release comes out.

**Comment by geoand** [comment]
Yeah, bumping versions usually brings some pain and I'll handle that.

Also, it's likely your stuff and the tool support will be released together

**Comment by edeandrea** [comment]
> Also, it's likely your stuff and the tool support will be released together

Then I guess we'll see what we get once a release happens :)

**Comment by geoand** [comment]
The new release is going out tomorrow.

I've already worked locally to have this repo compatible with it, so if you want to have this ready, I can include it in a release I plan to do tomorrow evening (my time)

**Comment by edeandrea** [comment]
So are you saying release this PR as its written today?

**Comment by edeandrea** [comment]
As it stands right now this PR was just waiting on https://github.com/langchain4j/langchain4j/pull/873 to be released. Once it it is available then this PR should compile fine and all the tests should run as expected. It could then be moved out of "Draft" and merged.

**Comment by geoand** [comment]
I'm saying that if you bring it to a state where everything works with tomorrow's release (ideally with tools as well but not absolutely necessary), then I'll include it

**Comment by geoand** [comment]
Tool support should be easy, you can check the commit for Mistral

**Comment by edeandrea** [comment]
As of right now that tools PR hasn't even been merged upstream. Still in draft status. I'll check in with this first thing when I get to my desk in the morning (which will be your afternoon).

**Comment by edeandrea** [comment]
Doesn't look like the release has happened yet

**Comment by geoand** [comment]
It's brewing and I'm working on the tools stuff in parallel

**Comment by geoand** [comment]
I have it working so when upstream release is done, I will update and force push to this PR

**Comment by edeandrea** [comment]
I guess that means that what I did here in this PR is good too :)

**Comment by geoand** [comment]
I had to make one small change, but other than that it worked as advertised :)

File: samples/email-a-poem/src/main/java/io/quarkiverse/langchain4j/sample/MyAiService.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import io.quarkiverse.langchain4j.RegisterAiService;
 
 @RegisterAiService(tools = EmailService.class)
+//@RegisterAiService
```
Comment (edeandrea) [inline samples/email-a-poem/src/main/java/io/quarkiverse/langchain4j/sample/MyAiService.java:8]: Should this have been included?

**Comment by edeandrea** [comment]
LGTM!

**Comment by geoand** [comment]
Ideally they would be isolated but I "need" to release today so I don't have the luxury of having 2 PRs

**Comment by edeandrea** [comment]
You're the boss!

**Pull request merged**